runbooks/workflow/creating-maintaining-littlepay-data-syncs.md

```diff
@@ -0,0 +1,51 @@
+
+You can confirm you have access to a given access key by listing the keys for the specific instance using the profile you configured:
+`aws iam list-access-keys --user-name <username> --profile <profile>`. **Note that the username may not be exactly the same as the `merchant_id`; check the key JSON or error output for the username.** For example, `An error occurred (AccessDenied) when calling the ListAccessKeys operation: User: arn:aws:iam::840817857296:user/system/sbmtd-default is not authorized to perform: iam:ListAccessKeys on resource: user sbmtd because no identity-based policy allows the iam:ListAccessKeys action` indicates that the username is `sbmtd-default`.
+
+### First-Time Setup Steps
+
+When you receive an access key from Littlepay for an agency that is being newly synced, you need to add that access key to Google Cloud Secret Manager, named with the format `LITTLEPAY_AWS_IAM_<MERCHANT_ID>_ACCESS_KEY` like the existing keys. Secret names are uppercase, and we have removed hyphens (or convert to underscore) in the `merchant_id` historically.
+
+Next, [create a service account in the cal-itp-data-infra project](https://console.cloud.google.com/iam-admin/serviceaccounts/create?walkthrough_id=iam--create-service-account&project=cal-itp-data-infra#step_index=1) specifically for this agency's Littlepay sync, using the naming convention `<MERCHANT_ID>-payments-user`. There is a pre-configured role titled Agency Payments Service User, which will give the new service account the correct permissions when assigned to that account.
+
+After that, you should create a simple YAML file in [the sync-littlepay DAG folder in the data-infra repository](https://github.com/cal-itp/data-infra/tree/8516a1a4ab2ecfe6ef33e3fbc4224bcedbd06e98/airflow/dags/sync_littlepay), structurally similar to the others that are there, utilizing the instance name and access key name for the agency you're setting up. Similarly, you should create a YAML file corresponding to the new agency in [the parse-littlepay DAG folder](https://github.com/cal-itp/data-infra/tree/8516a1a4ab2ecfe6ef33e3fbc4224bcedbd06e98/airflow/dags/parse_littlepay). Then, add a mapping between the agency's instance identifier and its Cal-ITP internal GTFS dataset identifier in [this seed file](https://github.com/cal-itp/data-infra/blob/main/warehouse/seeds/payments_gtfs_datasets.csv).
+
+[This pull request](https://github.com/cal-itp/data-infra/pull/2928/files) demonstrates all four of the code changes involved in this setup process (though the row-level access policy setting is now in a macro [here](https://github.com/cal-itp/data-infra/blob/main/warehouse/macros/create_row_access_policy.sql#L21), rather than in the `fct_payments_rides_v2` model).
+
+After these steps are completed, the new Littlepay account's data should start flowing the next time [the external table Airflow DAG](https://o1d2fa0877cf3fb10p-tp.appspot.com/dags/create_external_tables/grid) runs.
```

Comment:
This is a bit confusing to me; I don't understand why the external table would be the task to highlight. Don't all 4 Airflow DAG tasks (LP sync, LP parse, create external tables, transform warehouse) need to run before the data is available in the mart?

Reply:
You're right; I was in the headspace of the data already being present in GCS and external tables being run to get it reflected in the warehouse. But that's not the workflow - I'll update.

Reply:
Resolved in [89bc612](https://github.com/cal-itp/data-infra/pull/3019/commits/89bc6127e14d1cd9945f0b09fff2968eaec3bc9d)

Reply:
Thanks, I think the updates are good, only question to me is whether it's preferable to link to the Airflow UI or the Airflow DAG task README docs -- I'm fine if you want to leave it as-is but just figured I'd mention the alternative